### ISSUE TITLE
Validating xml-embedded certificates

### ISSUE THREAD

**Issue by tscheurer** [Sustainsys/Saml2#861]
While loading federation metadata, AuthServices can currently just validate the trust-chain of certificates added in SigningCertificates. Currently there is no possibility to validate the trust of the certificates embedded in the signed-xml.

According to msdn you need to extract the certificate manually and perform validation. I could  accomplish this with the following code snippet:

```csharp
var signedXml = new SignedXml(xmlMetaData);
XmlElement MessageSignatureNode = xmlMetaData.DocumentElement["Signature", SignedXml.XmlDsigNamespaceUrl];
signedXml.LoadXml(MessageSignatureNode);

var x509data = signedXml.Signature.KeyInfo.OfType<KeyInfoX509Data>().First();
foreach (X509Certificate2 certificate in x509data.Certificates)
{
	if (!certificate.Verify())
	{
		throw new Exception("Could not verify: " + certificate.FriendlyName);
	}
}
```
As long as I add the provided Root CA Certificat to my Trusted Root Certificate Authorities, this test passes successfully.

This would add security by just importing a long-lasting Root certificate.

**Comment by AndersAbel** [comment]
The SAML2 spec is rather clear on that using the certificate format is just a matter of using a standard format for transferring the key information. The trust is not in that the certificates are signed by an external certificate authority. The trust is in that the certificates are (as part of the metadata) signed by the key of the federation.

It's even common to have non-valid certificates published, that must be supported. So if you have a set up where you really need to validate the certificates in the federation, that must be optional. Maybe we could offer a notification when the certificate information has been parsed. At that point we can get an `X509Certificate2` instance. Please note though that there are other format for key representation that are not a certificate at all. They will of course not be possible to validate.

**Comment by tscheurer** [comment]
Our implementation is mostly used with the SWITCH edu-ID/AAI Federation. They do currently not provide the signing-key, besides of course it is included in the metadata. They instead provide their CA certificate, which they use to sign the signature certificate.

In this case it would be useful and I think it should be optional setting. 
I can understand that this is just one of many possibilities to establish trust.

**Comment by AndersAbel** [comment]
@tscheurer I see the use case then. But then the certificate embedded in the metadata should be checked against that particular root certificate. Validating it against all trusted roots on the machine is not enough - I could sign the metadata with my certificate for sustainsys.com and it would validate against a known trusted root.

**Comment by tscheurer** [comment]
@AndersAbel You are right, this is a good point. I would have to check it against the specific CA certificate, not just any trusted. Thx
